helpers/prism.js

```diff
@@ -0,0 +1,432 @@
+
+	.token.url { color: var(--d2l-code-token-url); }
+
+	.language-css .token.string,
+	.style .token.string,
+	.token.entity,
+	.token.operator,
+	.token.url {
+		background-color: transparent;
+	}
+
+	pre[class*="language-"].line-numbers {
+		counter-reset: linenumber;
+		padding-left: 3.8em;
+		position: relative;
+	}
+
+	pre[class*="language-"].line-numbers > code {
+		position: relative;
+		white-space: inherit;
+	}
+
+	.line-numbers .line-numbers-rows {
+		border-right: 1px solid var(--d2l-code-line-numbers-separator);
+		font-size: 100%;
+		left: -3.8em;
+		letter-spacing: -1px;
+		pointer-events: none;
+		position: absolute;
+		top: 0;
+		-webkit-user-select: none;
+		-moz-user-select: none;
+		-ms-user-select: none;
+		user-select: none;
+		width: 3em; /* works for line-numbers below 1000 lines */
+	}
+
+	.line-numbers-rows > span {
+		counter-increment: linenumber;
+		display: block;
+	}
+
+	.line-numbers-rows > span::before {
+		color: var(--d2l-code-line-numbers);
+		content: counter(linenumber);
+		display: block;
+		padding-right: 0.8em;
+		text-align: right;
+	}
+`;
+
+const getLanguageInfo = elem => {
+	const classes = elem.classList;
+	for (let i = 0; i < classes.length; i++) {
+		if (classes[i].startsWith('language-')) {
+			const key = classes[i].substring(9);
+			const desc = codeLanguages.get(key);
+			if (desc) return { key: key, desc: desc };
+		}
+	}
+	return { key: 'plain', desc: codeLanguages.get('plain') };
+};
+
+const languageDependencies = {
+	arduino: [ 'cpp' ],
+	cpp: [ 'c' ],
+	racket: [ 'scheme' ]
+};
+
+const languagesLoaded = {
+	clike: Promise.resolve(),
+	css: Promise.resolve(),
+	javascript: Promise.resolve(),
+	markup: Promise.resolve(),
+	plain: Promise.resolve()
+};
+
+const loadLanguage = async key => {
+	if (languagesLoaded[key]) return languagesLoaded[key];
+
+	// Prism languages can extend other anguages and must be loaded in order
+
+	// eslint-disable-next-line no-async-promise-executor
+	languagesLoaded[key] = new Promise(async resolve => {
+		if (languageDependencies[key]) {
+			await Promise.all(languageDependencies[key].map(dependencyKey => loadLanguage(dependencyKey)));
+		}
+
+		const script = document.createElement('script');
+		script.async = 'async';
+		script.onload = resolve;
+		script.src = `${prismLocation}/components/prism-${key}.min.js`;
+		document.head.appendChild(script);
+	});
+
+	return languagesLoaded[key];
+};
+
+const pluginsLoaded = {};
+
+const loadPlugin = async plugin => {
+	if (pluginsLoaded[plugin]) return pluginsLoaded[plugin];
+
+	pluginsLoaded[plugin] = new Promise(resolve => {
+		const script = document.createElement('script');
+		script.async = 'async';
+		script.onload = resolve;
+		script.src = `${prismLocation}/plugins/${plugin}/prism-${plugin}.min.js`;
+		document.head.appendChild(script);
+	});
+
+	return pluginsLoaded[plugin];
+};
+
+let prismLoaded;
+
+const loadPrism = () => {
+	if (prismLoaded) return prismLoaded;
+
+	prismLoaded = Promise.all([
```

Comment:
We use a `Promise.all` instead of sequential `await` calls so we can load the JS and CSS in parallel.

Reply:
Efficiency!